### PR TITLE
Introduce full promethues deployment.

### PR DESCRIPTION
**Pull request by mrsiano**
this patch creates new ansible playbook, to install prometheus.

### CODE REVIEW

**Comment by mrsiano** [comment]
@mbruzek please review

**Comment by jeremyeder** [comment]
Why would we carry a template here?  Can't we use the origin one and then customize it?

**Comment by jmencak** [comment]
In my view Prometheus should be installed using https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_prometheus role, which supports persistent volumes.

**Comment by mrsiano** [comment]
@jeremyeder totally agreed, thats why I put this comment https://github.com/mbruzek/ansible-openstack-setup/blob/9a786bd5ffb61c1f880b72730ed05586d44230d0/prometheus.yml#L17

since this one isn't merge yet, I place it here.

**Comment by mrsiano** [comment]
@jmencak @jeremyeder so we may consider to use it there openshift-ansible. please advise

File: prometheus-high-performance.yaml [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+- description: Time interval for scrape config (1m, 5s)
+  name: SCRAPE_INTERVAL
+  value: 15s
+objects:
+# Authorize the prometheus service account to read data about the cluster
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: prometheus
+    namespace: "${NAMESPACE}"
```
Comment (mbruzek) [inline prometheus-high-performance.yaml]: Where do these shell variables get resolved? This is an implementation detail I do not yet understand.

File: prometheus.yml [hunk 2]
```diff
@@ -5,3 +5,16 @@
...
+      mode: 755
+      with_items:
+        - { src: 'redeploy-prometheus.sh', dest: '~/' }
+        - { src: 'prometheus.yml', dest: '~/' }
+        - { src: 'prometheus-high-performance.yaml', dest: '~/' }
+
+    # TODO:// grab prometheus-high-performance.yaml directly from github.
+
+    - name: deploy promethues
+      shell: sh ~/prometheus-high-performance.yaml deploy
```
Comment (mbruzek) [inline prometheus.yml]: I am new to Ansible, but I have read that you use the `command` module over the `shell` module unless you need to do some pipe (|) or output redirection (>).

**Review by mbruzek — COMMENTED** [review]
Overall this looks good. @jeremyeder brings up a good point that if the prometheus template exists in the origin code we may be able to copy it from the location on the ocp-3.7 "gold" images. Earlier steps have to clone the origin repo to /root/origin so you could start looking for it there.

I have some questions about how the shell variables are replaced and the paths and mount points worried about those existing before running this script.

File: prometheus-high-performance.yaml [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        scheme: https
+
+        # This TLS & bearer token file config is used to connect to the actual scrape
+        # endpoints for cluster components. This is separate to discovery auth
+        # configuration because discovery & scraping are two separate concerns in
+        # Prometheus. The discovery auth config is automatic if Prometheus runs inside
+        # the cluster. Otherwise, more config options have to be provided within the
+        # <kubernetes_sd_config>.
+        tls_config:
+          ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
```
Comment (mbruzek) [inline prometheus-high-performance.yaml]: Does this directory need to exist? On which server are you targeting this?

File: prometheus-high-performance.yaml (hunk 1 above)
Comment (mrsiano) [reply prometheus-high-performance.yaml]: currently it is hard-coded based on the original origin code 
https://github.com/mbruzek/ansible-openstack-setup/blob/9a786bd5ffb61c1f880b72730ed05586d44230d0/prometheus-high-performance.yaml#L14

File: prometheus.yml (hunk 2 above)
Comment (mrsiano) [reply prometheus.yml]: ok, make sense we can replace it no problem.

File: prometheus-high-performance.yaml (hunk 3 above)
Comment (mrsiano) [reply prometheus-high-performance.yaml]: by default I didn't take any further action about his directory, but we need to make sure to run the shell script on the master node, otherwords deploy the prometheus from the master node.

File: prometheus-high-performance.yaml (hunk 1 above)
Comment (mbruzek) [reply prometheus-high-performance.yaml]: I see it now thank you!

**Comment by mbruzek** [comment]
We discussed this PR after our team meeting today and decided to wait until more of the pieces are together. I would ask that the `prometheus-high-performance.yaml` file be moved to the templates directory and the hosts target the master group or a single master of the cluster.

**Comment by mrsiano** [comment]
@jmencak @jeremyeder @mbruzek so i've play with it a little bit, I've got an ugly thing but, lets consider how to have nicer solution.

based on having ansible-openstack-setup repo, on my deployed cluster.

1. login to master node
2. git clone https://github.com/mbruzek/ansible-openstack-setup.git
3. cd ansible-openstack-setup
4. git checkout prometheus
5. oc new-app -f ./templates/prometheus-high-performance.yaml   # deploy prometheus
6. git checkout grafana 
7.  echo -e "[masters]\n`hostname`" > gf.ini
8. ansible-playbook -i gf.ini graphana.yml # deploy grafana

**Comment by jeremyeder** [comment]
Hang on -- @mrsiano you should assume prometheus is deployed via openshift-ansible.  openshift-ansible deploys a different template than the one you and I have been using.

We should *not* carry a "high performance template".  We should file RFEs against the openshift-ansible jinja templates to expose the knobs we need, so that we can use openshift-ansible to deploy prometheus how we want it.

https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_prometheus/templates

So, I am not sure if we need this PR anymore.  I don't think we do.

**Comment by mbruzek** [comment]
Based on the last comment from @jeremyeder I think we should pause/close this PR in favor of making a PR toward the openshift-ansible. The scale-ci automation runs the byo/config.yml playbook. If you tell me what variables to configure for Prometheus  I can start passing them in the `all.yml` or `OSEv3.yml` variable files . I am leaving the PR open to see if @mrsiano agrees or has more information on this approach.

**Pull request closed without merging**